lib/src/signed_video_h26x_auth.c

```diff
@@ -198,32 +196,38 @@ prepare_for_link_and_gop_hash_verification(signed_video_t *self, h26x_nalu_list_
     // At the start of the GOP, initialize the |crypto_handle| to enable
     // sequentially updating the hash with more NALUs.
     SV_THROW(openssl_init_hash(self->crypto_handle, true));
+    int num_i_frames = 0;
     // Iterate through the NALU list until the end of the current GOP or SEI item is found.
     while (item && !(found_next_gop || found_item_after_sei)) {
       // Skip non-pending items
       if (item->validation_status != 'P') {
         item = item->next;
         continue;
       }
-
+      num_i_frames += item->nalu->is_first_nalu_in_gop;
+      if (num_i_frames > 1) break;  // Break if encountered second I frame.
       // Ensure that only non-missing NALUs (which have non-null pointers) are processed.
       assert(item->nalu);
 
-      // Track when the current item follows the SEI or marks the beginning of the next GOP.
-      found_item_after_sei = (item->prev == sei);
-      found_next_gop = (item->nalu->is_first_nalu_in_gop && !item->need_second_verification);
-
       // Skip GOP SEI items as they do not contribute to the GOP hash.
       if (item->nalu->is_gop_sei) {
         item = item->next;
+        if (item) {
+          // Track when the current item follows the SEI or marks the beginning of the next GOP.
+          found_item_after_sei = (item->prev == sei);
+        } else {
+          found_next_gop = true;
+        }
         continue;
       }
-      hash_to_add = item->need_second_verification ? item->second_hash : item->hash;
+      hash_to_add = item->nalu->is_first_nalu_in_gop ? item->second_hash : item->hash;
       // Since the GOP hash is initialized, it can be updated with each incoming NALU hash.
       SV_THROW(openssl_update_hash(self->crypto_handle, hash_to_add, hash_size, true));
       item->used_in_gop_hash = true;  // Mark the item as used in the GOP hash
-
```

Comment:
Add back this empty line. It becomes more clear that the end of the while is reached and moved to the next item.

Reply:
Well, you added back an empty line **before** `item->used_in_gop_hash = true;` instead of **after**...